### PR TITLE
Make Chip component

### PR DESCRIPTION
**Pull request by ShrawanBK**

### CODE REVIEW

File: src/components/Chip/index.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                className={_cs(
+                    styles.action,
+                    actionClassName,
+                )}
+            >
+                {action}
+            </Button>
+        </span>
+    );
+}
```
Comment (frozenhelium) [inline src/components/Chip/index.tsx]: We do not need to separate this out

File: src/components/Chip/index.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                <span
+                    className={_cs(styles.icon, iconClassName)}
+                >
+                    {icon}
+                </span>
+            )}
+            {children ? { children } : (
+                <span
+                    className={_cs(
+                        styles.label,
```
Comment (frozenhelium) [inline src/components/Chip/index.tsx:137]: add `labelClassName` prop as well

File: src/components/Chip/styles.css [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        justify-content: center;
+        align-items: center;
+
+        .action {
+            margin: var(---tui-spacing-medium) var(--tui-spacing-large);
+            cursor: pointer;
+            color: var(--tui-color-text-disabled-light);
+            background-color: transparent;
+            height: 100%;
+            width: 100%;
```
Comment (frozenhelium) [inline src/components/Chip/styles.css]: we should not use `width`, `height` inside a flex, instead we can do `flex-grow: 1` for full width, and `align-self: stretch` for full height

File: src/components/Chip/styles.css [hunk 4]
```diff
@@ -0,0 +1,48 @@
+.chip-row {
```
Comment (frozenhelium) [inline src/components/Chip/styles.css]: The general layout for chip should be like this
```scss
.chip {
    display: flex;

    .icon {
        flex-shrink: 0;
    }
    .label {
        flex-grow: 1;
    }
    .action {
        flex-shrink: 0;
    }
}
```
```
-----------------------------------------------
| icon | label                       | action |
-----------------------------------------------
```

@safar @AdityaKhatri thoughts?

File: src/components/Chip/index.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
+import React, { ReactNode } from 'react';
+import { _cs } from '@togglecorp/fujs';
+
+import styles from './styles.css';
+import Button, { ButtonProps } from '../Button';
+
+export interface ActionComponentProps {
```
Comment (tnagorra) [inline src/components/Chip/index.tsx]: We don't need to export props for internal component

File: src/components/Chip/index.tsx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+                <span
+                    className={_cs(
+                        styles.label,
+                    )}
+                >
+                    {label}
+                </span>
+            )}
+            {action && (
+                <ActionComponent
```
Comment (tnagorra) [inline src/components/Chip/index.tsx]: Why have a separate component for `action` when we don't have separate components for `icon` and `label`?

File: src/components/Chip/styles.css (hunk 4 above)
Comment (tnagorra) [reply src/components/Chip/styles.css]: I agree

Comment (AdityaKhatri) [reply src/components/Chip/styles.css]: makes sense!

File: src/stories/Chip/Chip.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+export default {
+    title: 'View/Chip',
+    component: Chip,
+    argTypes: {},
+};
+
+const Template = (args: ChipProps) => (
+    <Chip
+        {...args}
+        label="default with acition"
```
Comment (AdityaKhatri) [inline src/stories/Chip/Chip.stories.tsx]: typo alert!

File: src/components/Chip/styles.css [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    .icon {
+        align-self: center;
+    }
+
+    .disabled-label {
+        color:var(--tui-color-text-disabled-dark)
+    }
+
+    .action-component {
+        height: var(--tui-font-size-extra-large);
```
Comment (AdityaKhatri) [inline src/components/Chip/styles.css]: Lets not do this, as action component has the possibility to contain more than one wider elements.
As Ankit suggested, make the label grow and don't allow the icon and action to shrink.

File: src/components/Chip/index.tsx [hunk 9]
```diff
@@ -99,27 +59,39 @@ function Chip(props: ChipProps) {
...
+                : (
+                    <span
+                        className={_cs(
+                            styles.label,
+                            labelClassName,
+                        )}
+                    >
+                        {label}
+                    </span>
+                )}
```
Comment (tnagorra) [inline src/components/Chip/index.tsx]: use `value ?? defaultValue` syntax

File: src/components/Chip/index.tsx [hunk 10]
```diff
@@ -50,13 +81,37 @@ function Chip(props: ChipProps) {
...
+    }, [variant]);
+
+    const themeClassName = useThemeClassName(innerUiMode, styles.light, styles.dark);
+    const innerThemeClassName = useThemeClassName(
+        innerUiMode,
+        styles.innerLight,
+        styles.innerDark,
+    );
+
+    console.log(innerUiMode);
```
Comment (tnagorra) [inline src/components/Chip/index.tsx]: No console.log on render

File: src/stories/Chip/Chip.stories.tsx [hunk 11]
```diff
@@ -29,6 +28,37 @@ Default.args = {};
 export const Variants = () => (
     <div className={styles.chipVariants}>
         <section>
+            <h3>Variants</h3>
```
Comment (tnagorra) [inline src/stories/Chip/Chip.stories.tsx]: Should create different story for different features

**Pull request merged**